control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go

```diff
@@ -70,7 +87,7 @@ func TestReconcileErrorHandling(t *testing.T) {
 	var totalCreates int
 	{
 		fakeClient := &testClient{
-			Client: fake.NewClientBuilder().WithScheme(api.Scheme).Build(),
+			Client: fake.NewClientBuilder().WithScheme(api.Scheme).WithObjects(initialObjects...).Build(),
```

Comment:
So the way the test works is that we have are tallying the number of get errors and matching them to the number of createOrUpdate failures so that we can detect whether we're short-circuiting the reconcile early. However, in this case, the createOrUpdate happens on the control plane side (to write the observed-config-blah) and so an error in reading the global config from the target cluster will not match an error to updateOrCreate on the same client.